src/components/Bullet/index.tsx

```diff
@@ -0,0 +1,133 @@
+const variants = {
+  danger: ({ theme }: { theme: Theme }) => `
+    background-color: ${theme.colors.danger.background};
+    color: ${theme.colors.danger.text};
+  `,
+  default: ({ theme }: { theme: Theme }) => `
+    border: 1px solid ${theme.colors.neutral.borderWeak};
+    background-color: ${theme.colors.neutral.background};
+    color: ${theme.colors.neutral.text};
+  `,
+  disabled: ({ theme }: { theme: Theme }) => `
+  border: 1px solid ${theme.colors.neutral.borderWeak};
+    background-color: ${theme.colors.neutral.background};
+    color: ${theme.colors.neutral.textWeak};
+  `,
+  on: ({ theme }: { theme: Theme }) => `
+    background-color: ${theme.colors.primary.background};
+    color: ${theme.colors.primary.text};
+  `,
+  success: ({ theme }: { theme: Theme }) => `
+    background-color: ${theme.colors.success.background};
+    color: ${theme.colors.success.text};
+  `,
+  wait: ({ theme }: { theme: Theme }) => `
+    background-color: ${theme.colors.info.background};
+    color: ${theme.colors.info.text};
+  `,
+  warning: ({ theme }: { theme: Theme }) => `
+    background-color: ${theme.colors.warning.background};
+    color: ${theme.colors.warning.text};
+  `,
+} as const
+type BulletVariant = keyof typeof variants
+export const bulletVariants = Object.keys(variants) as BulletVariant[]
+
+const sizes = {
+  medium: ({ theme }: { theme: Theme }) => css`
+    width: ${theme.space['4']};
+    height: ${theme.space['4']};
+    font-size: 16px;
+  `,
+  small: ({ theme }: { theme: Theme }) => css`
+    width: ${theme.space['3']};
+    height: ${theme.space['3']};
+    font-size: 14px;
+  `,
+} as const
+type BulletSize = keyof typeof sizes
+export const bulletSizes = Object.keys(sizes) as BulletSize[]
+
+const iconSizes: Record<BulletSize, number> = {
+  medium: 14,
+  small: 10,
+}
+
+const variantStyles = ({ variant }: { variant: BulletVariant }) =>
+  variants[variant]
+const sizeStyles = ({ size }: { size: BulletSize }) => sizes[size]
+
+type StyledContainerType = { variant: BulletVariant; size: BulletSize }
+const StyledContainer = styled('div')<StyledContainerType>`
+  display: inline-flex;
+  border-radius: 50%;
+  justify-content: center;
+  align-items: center;
+  font-weight: 500;
+  ${variantStyles}
+  ${sizeStyles}
+`
+
+type ContentProps =
+  | { icon: ComponentProps<typeof Icon>['name']; text?: never }
+  | { icon?: never; text: string }
+type BulletProps = {
+  className?: string
+  size?: BulletSize
+  tooltip?: string
+  tooltipBaseId?: string
+  variant?: BulletVariant
+} & ContentProps
+
+const Bullet: FunctionComponent<BulletProps> = ({
+  className,
+  variant = 'default',
+  size = 'medium',
+  icon,
+  text,
+  tooltip,
+  tooltipBaseId,
+}) => (
+  <Tooltip baseId={tooltipBaseId} text={tooltip}>
+    <StyledContainer variant={variant} size={size} className={className}>
+      {icon ? <Icon name={icon} size={iconSizes[size]} /> : text}
+    </StyledContainer>
+  </Tooltip>
+)
+
+const customXorPropType = (
```

Comment:
What if we put icon and text as optional ?

Reply:
it works fine but you have no Proptypes warning if both aren't specified

Reply:
Yeah, don't care about proptypes :D
Less logic we add the better